node/src/services/blockchain/handler.rs

```diff
@@ -330,6 +335,42 @@ impl BlockchainService {
                             size,
                             user_peer_ids: peer_ids,
                         }),
+                        RuntimeEvent::FileSystem(
+                            pallet_file_system::Event::AcceptedBspVolunteer {
+                                who,
+                                location,
+                                fingerprint,
+                                multiaddresses,
+                                owner,
+                                size,
+                            },
+                        ) => {
+                            // We try to convert the types coming from the runtime into our expected types.
+                            let fingerprint: Fingerprint = fingerprint.as_bytes().into();
+                            // In this case, the Multiaddresses come as a BoundedVec of BoundedVecs of bytes,
+                            // and we need to convert them. It only fails if the bytes do not represent a valid multiaddress.
+                            // TODO: We should make sure Multiaddresses are valid from inside the runtime.
```

Comment:
We have a TODO for this because we considered it, but it is not easy to do that check in a runtime with no std apparently.